controllers/humiocluster_controller_test.go

```diff
@@ -911,6 +918,15 @@ var _ = Describe("HumioCluster Controller", func() {
 			}, testTimeout, testInterval).Should(Succeed())
 
 			// TODO: Right now the service is not updated properly, so we delete it ourselves to make the operator recreate the service
+			// Wait for the new HumioCluster to finish any existing reconcile loop by waiting for the
+			// status.observedGeneration to equal at least that of the current resource version. This will avoid race
+			// conditions where the HumioCluster is updated and service is deleted mid-way through a reconcile.
+			resourceVersion, _ = strconv.Atoi(updatedHumioCluster.ResourceVersion)
+			Eventually(func() bool {
+				k8sClient.Get(ctx, key, &updatedHumioCluster)
+				observedGeneration, _ := strconv.Atoi(updatedHumioCluster.Status.ObservedGeneration)
+				return observedGeneration > resourceVersion
+			}, testTimeout, testInterval).Should(BeTrue())
```

Comment:
Perhaps we should just `return observedGeneration` then use `BeNumerically()` to compare the values? Then we'd be able to see an error stating what exact values `resourceVersion` and `observedGeneration` had when the test fails instead of just the fact that it isn't true. See more here: https://onsi.github.io/gomega/#benumericallycomparator-string-compareto-interface

Reply:
Ah I was looking for that type of helper but missed that one. Thanks!